### PR TITLE
Use `tomllib/tomli` over `toml`

### PR DESCRIPTION
**Pull request by AlexWaygood**
`tomli` is more actively maintained than `toml`, is more up to date with the TOML spec, and has been incorporated into the Python stdlib as `tomllib` in Python 3.11.

I think this is worth doing anyway, but this also helps with #282, as `tomli` is a `py.typed` package that ships with inline type annotations.

### CODE REVIEW

File: pyperformance/data-files/requirements.txt [hunk 1]
```diff
@@ -2,17 +2,13 @@
...
+psutil==5.9.5
     # via
     #   -r requirements.in
     #   pyperf
-pyparsing==3.0.8
-    # via packaging
 pyperf==2.6.0
     # via -r requirements.in
-toml==0.10.2
-    # via -r requirements.in
```
Comment (AlexWaygood) [inline pyperformance/data-files/requirements.txt:18]: I ran `pip-compile` to regenerate this file, and it also bumped `packaging` and `psutil`. Not sure if that's desirable or if I did something wrong there!

File: pyperformance/_pyproject_toml.py [hunk 2]
```diff
@@ -12,13 +12,18 @@
...
 import packaging.requirements
 import packaging.specifiers
 import packaging.utils
 import packaging.version
-import toml
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
```
Comment (corona10) [inline pyperformance/_pyproject_toml.py:26]: I prefer to use the ImportError strategy for this code.
We should consider alternative implementations and we can not guarantee the tomlib from those implementations.

Comment (AlexWaygood) [reply pyperformance/_pyproject_toml.py]: Thanks, done!

**Review by corona10 — APPROVED** [review]
lgtm!

**Pull request merged**